cli/dts/lib.deno.unstable.d.ts

```diff
@@ -524,7 +530,10 @@ declare namespace Deno {
     rootName: string,
     sources?: Record<string, string>,
     options?: CompilerOptions,
```

Comment:
Removing `Deno.transpileOnly()` in favor of `noCheck` option means we need to change `options?: CompilerOptions` into:
```
options?: {
  noCheck?: boolean,
  compilerOptions?: CompilerOptions
}
```

Reply:
I suspect there are other non-tsconfig related options that we will want to make in here as well, like import map, so yeah, let's change it.